src/coreclr/src/jit/codegenxarch.cpp

```diff
@@ -1982,12 +1989,53 @@ void CodeGen::genMultiRegStoreToLocal(GenTree* treeNode)
             }
 
             assert(reg != REG_NA);
-            GetEmitter()->emitIns_S_R(ins_Store(type), emitTypeSize(type), reg, lclNum, offset);
-            offset += genTypeSize(type);
+            regNumber varReg = REG_NA;
+            if (isMultiRegVar)
+            {
+                regNumber  varReg      = treeNode->GetRegByIndex(i);
+                unsigned   fieldLclNum = varDsc->lvFieldLclStart + i;
+                LclVarDsc* fieldVarDsc = compiler->lvaGetDesc(fieldLclNum);
+                var_types  type        = fieldVarDsc->TypeGet();
+                if (varReg != REG_NA)
+                {
+                    hasRegs = true;
+                    if (varReg != reg)
+                    {
+                        inst_RV_RV(ins_Copy(type), varReg, reg, type);
+                    }
+                    fieldVarDsc->SetRegNum(varReg);
+                }
+                else
+                {
+                    if (!treeNode->AsLclVar()->IsLastUse(i))
+                    {
+                        GetEmitter()->emitIns_S_R(ins_Store(type), emitTypeSize(type), reg, fieldLclNum, 0);
+                    }
+                    fieldVarDsc->SetRegNum(REG_STK);
+                }
+            }
+            else
+            {
+                GetEmitter()->emitIns_S_R(ins_Store(type), emitTypeSize(type), reg, lclNum, offset);
+                offset += genTypeSize(type);
+            }
+        }
+        if (isMultiRegVar)
+        {
+            if (hasRegs)
+            {
+                genProduceReg(treeNode);
+            }
+            else
+            {
+                genUpdateLife(treeNode);
```

Comment:
This is the case where the fields were register candidates but happened to be spilled. We can't call `genProduceReg` in that case, as no registers are defined. However, in either this case, or the case where we have a destination that's not a candidate, we will generate multiple moves because the source has multiple registers, each of which must be either copied or stored.